infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/ghproxy-statefulset.yaml

```diff
@@ -0,0 +1,62 @@
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  namespace: default
+  name: ghproxy
+  labels:
+    app: ghproxy
+spec:
+  selector:
+    matchLabels:
+      app: ghproxy
+  replicas: 3
+  serviceName: ghproxy-hs
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: ghproxy
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: dedicated
+                    operator: In
+                    values:
+                      - "ghproxy"
+              topologyKey: kubernetes.io/hostname
+      containers:
+        - name: ghproxy
+          image: gcr.io/k8s-prow/ghproxy:v20200602-84dbfa37c9
+          args:
+            - --cache-dir=/cache
+            - --cache-sizeGB=99
+          # TODO: uncomment once gateway is migrated
+          # - --push-gateway=pushgateway
+          # - --serve-metrics=true
+          ports:
+            - containerPort: 8888
+          volumeMounts:
+            - name: ghproxy
+              mountPath: /cache
+      # run on our dedicated nodes
+      tolerations:
+        - key: "dedicated"
+          operator: "Equal"
+          value: "ghproxy"
+          effect: "NoSchedule"
+      nodeSelector:
+        dedicated: "ghproxy"
+  volumeClaimTemplates:
+    - metadata:
+        name: ghproxy
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 100Gi
+        storageClassName: gce-ssd-retain
```

Comment:
Why do we want to retain these PVs after they have been released?   Are we under some litigation-hold or do we do some post-facto analysis?

Reply:
I'm not sure about that, but I guess it prevents the cache from being cleared when ghproxy is upgraded by k8s-ci-robot.

Reply:
I think it's so we can avoid having to warm the cache if we have to kill/reschedule the ghproxy instance, @cjwagner does that sound right to you?

Reply:
That's not what retain means.  As long as you don't delete a PVC, the PV is yours to use and reuse.  Once you delete the PVC, the retention policy kicks in.  If you say "retain", you will "leak" the underlying disk until someone comes along and manually deletes it.

Reply:
Tim is right, retain shouldn't be used here. That was originally included because I misunderstood that concept.
Other Prow instances just use the default storage class (non-SSD) and it hasn't been a problem. If we want an SSD we'll still need to define a storage class, but it should not include `reclaimPolicy: Retain`.

Reply:
@cjwagner @thockin thank you for your explanations!
I updated the PR. the PVC will use the default `StorageClass`.